### PR TITLE
feat(discover): Add sidebar icon behind feature flag

### PR DESCRIPTION
**Pull request by lynnagara**
Add organization:discover feature flag and sidebar icon

Related: https://github.com/getsentry/getsentry/pull/2132

### CODE REVIEW

File: src/sentry/static/sentry/app/components/sidebar/index.jsx [hunk 1]
```diff
@@ -194,6 +197,15 @@ class Sidebar extends React.Component {
                   label={t('Projects')}
                   to={`/${organization.slug}/`}
                 />
+                {hasDiscoverFeature && (
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/sidebar/index.jsx]: Can we use the `<Feature>` component?

```
<Feature
feature={['discover']}
>
                  <SidebarItem
                    {...sidebarItemProps}
                    onClick={this.hidePanel}
                    icon={<InlineSvg src="icon-search" />}
                    label={t('Discover')}
                    to={`/organizations/${organization.slug}/discover/`}
                  />
</Feature>
```

File: src/sentry/static/sentry/app/views/organizationDiscover/index.jsx [hunk 2]
```diff
@@ -49,7 +49,7 @@ const OrganizationDiscoverContainer = createReactClass({
 
   render() {
     const {isLoading} = this.state;
-    const hasFeature = this.getFeatures().has('internal-catchall');
+    const hasFeature = this.getFeatures().has('discover');
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDiscover/index.jsx]: It'd be nice to use `<Feature>` here too, I think you can do:

```
<Feature feature={['discover']} renderNoFeatureMessage={this.renderComingSoon}>
```

**Comment by lynnagara** [comment]
Updated the sidebar, gonna leave the other one as-is since it's simpler/doesn't work exactly as you've proposed

**Review by billyvg — APPROVED** [review]
Curious what's wrong with `Feature` in the other component?

**Comment by lynnagara** [comment]
I think you can't give it `renderNoFeatureMessage` in that way

**Comment by billyvg** [comment]
Were you invoking it? It takes a function.

**Comment by lynnagara** [comment]
I wasn't, used exactly your snippet above

**Comment by billyvg** [comment]
Ah I didn't include the full snippet, does this work?

```
return (
<Feature feature={['discover']} renderNoFeatureMessage={this.renderComingSoon}>
      <div className="organization-home">
        {isLoading ? (
          this.renderLoading()
        ) : (
          <Discover
            organization={this.getOrganization()}
            queryBuilder={this.queryBuilder}
          />
        )}
      </div>
</Feature>
);

**Pull request merged**